contracts/permit/EIP2612PermitAndDeposit.sol

```diff
@@ -15,41 +40,60 @@ contract EIP2612PermitAndDeposit {
     /**
      * @notice Permits this contract to spend on a user's behalf, and deposits into the prize pool.
      * @dev The `spender` address required by the permit function is the address of this contract.
-     * @param _token Address of the EIP-2612 token to approve and deposit.
-     * @param _owner Token owner's address (Authorizer).
-     * @param _amount Amount of tokens to deposit.
-     * @param _deadline Timestamp at which the signature expires.
-     * @param _v `v` portion of the signature.
-     * @param _r `r` portion of the signature.
-     * @param _s `s` portion of the signature.
-     * @param _prizePool Address of the prize pool to deposit into.
-     * @param _to Address that will receive the tickets.
+     * @param _token Address of the EIP-2612 token to approve and deposit
+     * @param _owner Token owner's address (Authorizer)
+     * @param _amount Amount of tokens to deposit
+     * @param _deadline Timestamp at which the signature expires
+     * @param _permitSignature Permit signature
+     * @param _delegateSignature Delegate signature
+     * @param _prizePool Address of the prize pool to deposit into
+     * @param _to Address that will receive the tickets
+     * @param _ticket Address of the prize pool ticket
+     * @param _delegate The address to delegate the prize pool tickets to
      */
-    function permitAndDepositTo(
+    function permitAndDepositToAndDelegate(
         address _token,
         address _owner,
         uint256 _amount,
         uint256 _deadline,
-        uint8 _v,
-        bytes32 _r,
-        bytes32 _s,
+        PermitSignature calldata _permitSignature,
+        DelegateSignature calldata _delegateSignature,
         address _prizePool,
```

Comment:
Seems to me both the `_token` and `_ticket` can be pulled from the PrizePool itself.  It would add a few thousand gas, but make it a much simpler interface

Reply:
Done: https://github.com/pooltogether/v4-core/pull/249/commits/769cfda70b08e8cc0304de02483378212c785f56